### PR TITLE
prod_12232020_ESQueryChange

### PR DESCRIPTION
**Pull request by pegahtah**

### CODE REVIEW

File: src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java [hunk 1]
```diff
@@ -340,29 +343,35 @@ public void orderCancellation(ArrayList<KitRequest> kitRequests) throws Exceptio
...
-                            if (ddpParticipant != null) {
-                                kitRequests.add(new KitRequest(rs.getString(DBConstants.DSM_KIT_REQUEST_ID), rs.getString(DBConstants.DDP_PARTICIPANT_ID),
-                                        null, null, rs.getString(DBConstants.EXTERNAL_ORDER_NUMBER), ddpParticipant,
-                                        rs.getString(DBConstants.EXTERNAL_ORDER_STATUS),
-                                        rs.getString("subkits." + DBConstants.EXTERNAL_KIT_NAME),
-                                        rs.getLong(DBConstants.EXTERNAL_ORDER_DATE)));
-                            }
-                            else {
-                                logger.error("Participant not found in ES! " + rs.getString(DBConstants.DDP_PARTICIPANT_ID));
+                        String ddpParticipantId = rs.getString("ddp_participant_id");
```
Comment (SimoneMaiwald) [inline src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java]: use DBConstants.DDP_PARTICIPANT_ID please

File: src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java [hunk 2]
```diff
@@ -507,6 +516,22 @@ public static Map getParticipants(String realm) {
         return participantsESData;
     }
 
+    public static Map getParticipants(String realm, String ddpParticipantId) {
+        DDPInstance ddpInstance = DDPInstance.getDDPInstanceWithRole(realm, DBConstants.NEEDS_NAME_LABELS);
```
Comment (SimoneMaiwald) [inline src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java]: why do you get the ddpInstance with that role?

Comment (SimoneMaiwald) [reply src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java]: you could just pass in the instance!

File: src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java [hunk 3]
```diff
@@ -507,6 +516,22 @@ public static Map getParticipants(String realm) {
...
+        Map<String, Map<String, Object>> participantsESData = null;
+        String filter = "AND profile.guid = '";
+
+        if (StringUtils.isNotBlank(ddpInstance.getParticipantIndexES())) {
+            StringBuilder builder = new StringBuilder();
+            builder.append(filter);
+            builder.append(ddpParticipantId);
+            builder.append("'");
+            String queryFiler = builder.toString();
+            participantsESData = ElasticSearchUtil.getFilteredDDPParticipantsFromES(ddpInstance, queryFiler);
```
Comment (SimoneMaiwald) [inline src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java]: just use it like ElasticSearchUtil.getFilteredDDPParticipantsFromES(ddpInstance, ElasticSearchUtil.BY_GUID + ddpParticipantId). no need for StringBuilder

File: src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java [hunk 4]
```diff
@@ -340,29 +343,35 @@ public void orderCancellation(ArrayList<KitRequest> kitRequests) throws Exceptio
...
-                                        rs.getLong(DBConstants.EXTERNAL_ORDER_DATE)));
-                            }
-                            else {
-                                logger.error("Participant not found in ES! " + rs.getString(DBConstants.DDP_PARTICIPANT_ID));
+                        String ddpParticipantId = rs.getString("ddp_participant_id");
+                        logger.info("qyerying ES for ddpParticipantId: "+ddpParticipantId);
+                        if (StringUtils.isNotBlank(ddpParticipantId)) {
+                            Map participantsESData = getParticipants(ddpInstance.getName(), ddpParticipantId);
+                            if (participantsESData != null && !participantsESData.isEmpty()) {
+                                DDPParticipant ddpParticipant = ElasticSearchUtil.getParticipantAsDDPParticipant(participantsESData, rs.getString(DBConstants.DDP_PARTICIPANT_ID));
```
Comment (SimoneMaiwald) [inline src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java]: you already have the ddp_participant_id at that point, please reuse it

File: src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java [hunk 5]
```diff
@@ -507,6 +516,22 @@ public static Map getParticipants(String realm) {
         return participantsESData;
     }
 
+    public static Map getParticipants(String realm, String ddpParticipantId) {
```
Comment (SimoneMaiwald) [inline src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java]: rename it to getParticipant

File: src/test/java/org/broadinstitute/dsm/GBFTest.java [hunk 6]
```diff
@@ -155,4 +160,62 @@ public void confirmationGBFKit() throws Exception {
             Assert.fail("No apiKey found");
         }
     }
+
+
+    @Test
+    @Ignore
+    public void inputConfirmationGBFKit() throws Exception {
```
Comment (zyme) [inline src/test/java/org/broadinstitute/dsm/GBFTest.java:167]: @zyme here is the test

**Pull request closed without merging**